snmp/tests/test_e2e_core_profiles/test_profile_netgear_readynas.py

```diff
@@ -0,0 +1,126 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import pytest
+
+from datadog_checks.dev.utils import get_metadata_metrics
+
+from .. import common
+from ..test_e2e_core_metadata import assert_device_metadata
+from .utils import (
+    assert_common_metrics,
+    assert_extend_generic_if,
+    assert_extend_generic_ucd,
+    create_e2e_core_test_config,
+    get_device_ip_from_config,
+)
+
+pytestmark = [pytest.mark.e2e, common.py3_plus_only, common.snmp_integration_only]
+
+
+def test_e2e_profile_netgear_readynas(dd_agent_check):
+    config = create_e2e_core_test_config('netgear-readynas')
+    aggregator = common.dd_agent_check_wrapper(dd_agent_check, config, rate=True)
+
+    ip_address = get_device_ip_from_config(config)
+    common_tags = [
+        'snmp_profile:netgear-readynas',
+        'snmp_host:netgear-readynas.device.name',
+        'device_namespace:default',
+        'snmp_device:' + ip_address,
+    ] + []
+
+    # --- TEST EXTENDED METRICS ---
+    assert_extend_generic_if(aggregator, common_tags)
+    assert_extend_generic_ucd(aggregator, common_tags)
+    # --- TEST METRICS ---
+    assert_common_metrics(aggregator, common_tags)
+
+    # aggregator.assert_metric('snmp.netgear.nasMgrSoftwareVersion', metric_type=aggregator.GAUGE, tags=common_tags)
+    tag_rows = [
+        [
+            'netgear_disk_id:forward their oxen their acted driving',
+            'netgear_disk_interface:Jaded but kept their Jaded driving their kept',
+            'netgear_disk_model:acted forward kept forward',
+            'netgear_disk_serial:quaintly their',
+            'netgear_disk_slot_name:driving forward but oxen kept oxen oxen forward forward',
+        ],
+        [
+            'netgear_disk_id:kept forward oxen acted oxen acted Jaded',
+            'netgear_disk_interface:zombies but driving kept zombies acted their',
+            'netgear_disk_model:kept their driving Jaded oxen forward quaintly quaintly forward',
+            'netgear_disk_serial:but zombies forward forward but quaintly',
+            'netgear_disk_slot_name:their quaintly zombies quaintly kept quaintly their quaintly acted',
+        ],
+    ]
+    for tag_row in tag_rows:
+        aggregator.assert_metric('snmp.netgear.ataError', metric_type=aggregator.GAUGE, tags=common_tags + tag_row)
+        # aggregator.assert_metric(
+        #     'snmp.netgear.diskCapacity', metric_type=aggregator.GAUGE, tags=common_tags + tag_row
+        #     )
```

Comment:
this metric is of type `DisplayString` but according to the description it represents "The capacity of the disk in bytes" which means that values are numeric but sent in as string (o for example '1024' instead of 1024
you need to change the values in the snmprec with numerical values instead of the given string values and the test should pass